### PR TITLE
Work

### PR DESCRIPTION
**Pull request by ritasumtsova**
1. Fixed previous comments;
2. Added the Redux stuff to the Department & Login component;
3. Combine departments & department selectors into one.

### CODE REVIEW

File: src/components/pages/Departments/Departments.tsx [hunk 1]
```diff
@@ -1,33 +1,46 @@
...
-  const errMsg = useSelector((state: rootState) => state.departments.errorMsg);
-  // const loading = useSelector((state: rootState) => state.departments.loading);
+  const {
+    departmentsInfo:
+      {
+        loading,
+        departments,
+        errorMessage
+      }
+  } = useSelector(departmentsSelector);
```
Comment (eugenekovaljov) [inline src/components/pages/Departments/Departments.tsx:22]: please split such expressions into 2 steps.

File: src/store/actionCreators/thunks/departmentsThunks.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    try {
+      const res: AxiosResponse<Department> = await DepartmentsAPI.getDepartmentInfo(id);
+      const department: Department = res.data;
+
+      dispatch(fetchDepartmentByIdSuccess(department));
+    } catch (err) {
+      const error = err as AxiosError;
+      const errorMessage: string = error.message;
+
+      dispatch(fetchDepartmentByIdFailure(errorMessage));
```
Comment (eugenekovaljov) [inline src/store/actionCreators/thunks/departmentsThunks.ts:39]: you better do that in action creator

File: src/components/pages/Departments/Departments.tsx (hunk 1 above)
Comment (didevgen) [inline src/components/pages/Departments/Departments.tsx]: better split it into 2 expressions

File: src/store/actionCreators/thunks/authThunks.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    try {
+      const res: AxiosResponse<Auth> = await AuthAPI.login(userName, password);
+      const auth: Auth = res.data;
+
+      dispatch(loginSuccess(auth));
+    } catch (err) {
+      const error = err as AxiosError;
+      const errorMessage: string = error.message;
+
+      dispatch(loginFailure(errorMessage));
```
Comment (didevgen) [inline src/store/actionCreators/thunks/authThunks.ts:20]: put it into action creators

**Review by didevgen — APPROVED** [review]

File: src/store/actionCreators/thunks/authThunks.ts (hunk 3 above)
Comment (ritasumtsova) [reply src/store/actionCreators/thunks/authThunks.ts:20]: you mean to put the login thunk?

**Pull request merged**